builtin/logical/mysql/secret_creds.go

```diff
@@ -59,27 +81,44 @@ func (b *backend) secretCredsRevoke(
 		return nil, err
 	}
 
+	// Get the role name
+	roleNameRaw, ok := req.Secret.InternalData["role"]
+	roleName, ok := roleNameRaw.(string)
+
+	// init role entry struct
+	var role = &roleEntry{}
+	if roleName != "" {
+		role, err = b.Role(req.Storage, roleName)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	// Check for an empty revokeSQL string
+	// set it to a default query if the string is empty
+	if role.RevokeSQL == "" {
+		role.RevokeSQL = defaultRevokeSQL
+	}
+
 	// Start a transaction
 	tx, err := db.Begin()
 	if err != nil {
 		return nil, err
 	}
 	defer tx.Rollback()
 
-	// Revoke all permissions for the user. This is done before the
-	// drop, because MySQL explicitly documents that open user connections
-	// will not be closed. By revoking all grants, at least we ensure
-	// that the open connection is useless.
-	_, err = tx.Exec("REVOKE ALL PRIVILEGES, GRANT OPTION FROM '" + username + "'@'%'")
-	if err != nil {
-		return nil, err
-	}
+	for _, query := range strutil.ParseArbitraryStringSlice(role.RevokeSQL, ";") {
+		query = strings.TrimSpace(query)
+		if len(query) == 0 {
+			continue
+		}
+
+		query = strings.Replace(query, "{{name}}", username, -1)
```

Comment:
There was a comment in an earlier version that explained why this replacement was necessary...can you put a comment back, so that anyone thinking of pulling it out in the future understands why it's there? I think maybe this is part of the block that got moved up above?